packages/repository-json-schema/src/build-schema.ts

```diff
@@ -436,7 +436,19 @@ export function modelToJsonSchema<T extends object>(
 
     const propSchema = getJsonSchema(referenceType, options);
 
-    includeReferencedSchema(referenceType.name, propSchema);
+    if (typeof result.properties[p] !== 'boolean') {
```

Comment:
why are we testing if it's not equal to `boolean` specifically?

Reply:
JSON schema draft 6 allows a boolean for the property schema. This test skips that case so that we can safely cast the schema to `JSONSchema`.

Reply:
@raymondfeng , I see. Please add a comment indicating the reason to help maintainers understand that particular check.